src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
Ok, so [rfc2732](https://tools.ietf.org/html/rfc2732) (indirectly) specifies that the `Host:` header can contain an ipv6 address formatted with square brackets, so this is possible.

My solution would be to count bytes from the end of the string, checking if they are `'0'.ord <= byte <= '9'.ord`, breaking if they are not, then if you rech a `':'` character, note it's index then call `unsafe_byte_slice_string`. This would then be equivalent to the original regexp.